ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyRpcService.java

```diff
@@ -167,7 +168,14 @@ public void closeImpl() throws IOException {
 
   @Override
   public InetSocketAddress getInetSocketAddress() {
-    return (InetSocketAddress)getChannel().localAddress();
+    try {
+      return (InetSocketAddress) getChannel().localAddress();
+    } catch (IllegalStateException e) {
+      if (socketAddress.getPort() != NettyConfigKeys.Server.PORT_DEFAULT) {
+        return socketAddress;
+      }
```

Comment:
When `socketAddress.getPort() == NettyConfigKeys.Server.PORT_DEFAULT)`, why not returning `socketAddress`?

Reply:
Because if the port is PORT_DEFAULT which is 0, it's not a real port, cannot be returned to construct the RaftPeer object.

Reply:
I see.  Thanks.